### PR TITLE
fix(DB/Zone):improved Nagrand

### PR DESCRIPTION
**Pull request by Knindzagxg**
## Changes Proposed:
- Missing SAI creature updates.
- Improvements to existing if possible.

## Tests Performed:
- Applied SQL without problems.
- Tested and changed few due overlap in their cast time.

## How to Test:
- Apply SQL(s).
- Restart Server (You can .reload creature_template for each creature as well, but will take time)
- Go to zone (Check PR name) and check creatures around - You can also spawn (.npc add) each by command.

## Source for Changes:
- [Wowhead](https://wowhead.com) Database and or comments.
- [WotlkDB](https://wotlkdb.com/) Multi checking trough Private DB's.
- [WowWiki](https://wowwiki-archive.fandom.com/wiki/Portal:Main) Can be useful for extra informations.
- [WowPedia](https://wow.gamepedia.com/Wowpedia) Cross checking trough comments if possible.
- [Retail](https://worldofwarcraft.com/en-gb/) In-game checking if previous don't have any required informations.

## Target Branch:
- [x] Master

## Operating System:
- Linux (Debian 10)

## How to Test AzerothCore PR's
When a PR is ready to be tested, it will be marked as **[Waiting to be tested]**.
You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:
http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1616252519696041200.sql [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+(18205, 0, 0, 0, 2, 0, 100, 1, 20, 80, 0, 0, 11, 32023, 32, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Clefthoof - Between 20-80% Health - Cast \'32023\' (No Repeat)');
+
+UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 17130;
+DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 17130);
+INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
+(17130, 0, 0, 0, 0, 0, 100, 0, 2500, 5000, 20000, 25000, 11, 32019, 32, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Talbuk Stag - In Combat - Cast \'32019\'');
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1616252519696041200.sql:257]: It already has AI

Comment (pklloveyou) [reply data/sql/updates/pending_db_world/rev_1616252519696041200.sql:257]: I don't know if it's better than it is now

**Comment by pklloveyou** [comment]
First of all, it's better to have AI than none. I'll list individual creatures that have AI.

**Comment by FrancescoBorzi** [comment]
@Knindzagxg @pklloveyou please specify whether you have tested these changes in game, so we can merged them

File: data/sql/updates/pending_db_world/rev_1616252519696041200.sql (hunk 1 above)
Comment (Knindzagxg) [reply data/sql/updates/pending_db_world/rev_1616252519696041200.sql:257]: Changed cause Gore duration was overlaping.

**Comment by pklloveyou** [comment]
I simply tested it on each NPC to see if they would cause errors to the server

**Comment by pklloveyou** [comment]
In the original script, it also links 1713000

**Comment by claudiodfc** [comment]
This test is useless, SAI doesn't crash the server.

**Comment by Knindzagxg** [comment]
Any @azerothcore/testers for this?

**Comment by Shybuia** [comment]
I will do it tomorrow, if it would be too much, I will give partial report.

**Comment by Knindzagxg** [comment]
Not in rush, can do few a day and post when u done with all xP

**Comment by Shybuia** [comment]
TEST IN PROGRESS (will edit this comment tomorrow with rest results)

Used fast method of testing PR DB-only changes by [https://www.azerothcore.org/wiki/how-to-test-db-only-changes](url)

Video Playlist of State Before PR: https://www.youtube.com/playlist?list=PLNL5IEf0q3TGTXggNpfvb319oBYiZcSfj

State after PR:
https://youtu.be/lIkDYvanus8 Boulderfist Hunter now does spell Spear Throw on aggro. 
https://youtu.be/hPfB21KCkmw Gava'xi now does Gouge (once, 20-80% hp) and Sinister Strike in combat. 
https://youtu.be/Rqm-V0Si_Lg Tusker is applying Gore in combat and does Trample (once, 30-60%hp) and Stomp (once, 20-80%hp) 
https://youtu.be/hnq38RBRJXA Bull Elekk now does Trample (once, 20-80%hp) 
https://youtu.be/OEv9VQO4eA0 Subjugator Vaz'shir now does Trash in combat and Whirlwind (once, 20-80%hp) 
https://youtu.be/5Gzi1PvfofA Terrorguard now does Cripple in combat and War Stomp (once, 30-60%hp)
https://youtu.be/DVs2950A2mU Agitated Orc Spirit now does Dazed (once, 15-30%hp)
https://youtu.be/mud-2HKxZ14 Mountain Gronn now does Trash in combat.
https://youtu.be/7G6KZEQwy5o Watoosun's Polluted Essence casts now Acidic Splash in combat and Lake Surger casts Surge on aggro.
https://youtu.be/8IZbLBqyohY Boulderfist Warrior now cast Shoulder Charge on aggro and is dualwielding on respawn.

**Comment by Knindzagxg** [comment]
Perfect mate xP
I will wait till u finish everything.

Tho, no need to record everything if you need to save time.
You could go for those that have issues or i made mistake fixing xP

**Comment by Shybuia** [comment]
@Knindzagxg test completed yesterday, can you please reply to my previous comment if possible ? Thank you

**Comment by Shybuia** [comment]
@Knindzagxg after discussion with Sedatyf I edited my report because I missunderstood daze mechanic. 
Only 1 bug is present then: 
NOK https://youtu.be/5_wmRpuVLLc Mo'arg Master Planner does not cast Breath (18567) in combat

**Comment by Sedatyf** [comment]
Awesome work @Shybuia! @Knindzagxg you have a little red to fix and we're good to go 😄

**Comment by Knindzagxg** [comment]
Awesome mate, ill check everything later tonight or tomz.

Also, need to check about Thrash, cause yet again i think it might be added in mini pre Cata patch.

**Comment by wetbrownsauce** [comment]
> @Knindzagxg after discussion with Sedatyf I edited my report because I missunderstood daze mechanic.
> Only 1 bug is present then:
> 🟥 https://youtu.be/5_wmRpuVLLc Mo'arg Master Planner **does not cast Breath (18567) in combat**

this is not correct. You have mixed up the spell id with the npc id. The script says that NPC 18567 (mo'arg master planner) should cast SPELL 31705 (magnetic pull).

**Comment by Knindzagxg** [comment]
```UPDATE `creature_template` SET `AIName` = 'SmartAI' WHERE `entry` = 18567;
DELETE FROM `smart_scripts` WHERE (source_type = 0 AND entryorguid = 18567);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(18567, 0, 0, 0, 0, 0, 100, 0, 2500, 5000, 10000, 15000, 11, 31705, 32, 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 'Mo\'arg Master Planner - In Combat - Cast \'31705\'');```

He doesn't have breath, correct spell ID is used.
Can u re-test this one? It should pull you towards him if you using range attacks.

**Comment by wetbrownsauce** [comment]
Okay so this took about 4 hours so hopefully it will be useful information to someone. Here is my data on almost every mob in this PR as they are in TBC-Classic Live Servers.  

I tested each mob for ~2min - ~6min depending on the mob and number of mechanics they had. Here are the results.  

### [Windyreed Scavenger](https://tbc.wowhead.com/npc=17139/windyreed-scavenger)

- Spams Throw if not in melee range. Otherwise, uses melee. 
- Uses its Net ability in combat. Most commonly when a player is in melee range. Once netted, the mob moves out of melee range and begins using Throw again.  

### [Windyreed Wretch](https://tbc.wowhead.com/npc=17141/windyreed-wretch)
- Uses Creeping Mold every ~15sec until successfully applied. Once applied, only refreshes when the debuff expires.
- If Creeping Mold is dispelled, it will switch back to the ~15sec usage window.  
- Uses Diseased Touch every ~10-15sec regardless of HP%.

### [Clefthoof](https://tbc.wowhead.com/npc=18205/clefthoof)
- Casts Stomp after ~13sec of combat. Recasts every 15-50sec.

### [Talbuk Stag](https://tbc.wowhead.com/npc=17130/talbuk-stag)
- Casts Gore after ~7sec, recasts every 30sec.

### [Wild Elekk](https://tbc.wowhead.com/npc=18334/wild-elekk)
- Casts Gore after ~7sec, recasts every 30sec.

### [Talbuk Thorngrazer](https://tbc.wowhead.com/npc=17131/talbuk-thorngrazer)
- Uses Talbuk Strike after ~10sec, recast every 20-50sec
- Uses Gore after ~10sec., recast on expire (~15sec)

### [Windroc](https://tbc.wowhead.com/npc=17128/windroc)
- Casts Eagle Claw after ~15sec, refresh when debuff expires. 

### [Clefthoof Bull](https://tbc.wowhead.com/npc=17132/clefthoof-bull)
- Uses Hoof Stomp after ~13sec, recast every 30-40sec.

### [Tusker](https://tbc.wowhead.com/npc=18290/tusker)
- Uses Gore after ~10sec, recast on expire.
- Uses Trample every ~10sec.
- NEVER used Stomp. It is listed on WoWhead so idk why I never saw it in the 6min I fought for.

### [Storm Rager](https://tbc.wowhead.com/npc=17159/storm-rager)
- Cast Call Lightning after ~9sec, recast every 20sec.

### [Dust Howler](https://tbc.wowhead.com/npc=17158/dust-howler)
- Casts Howling Dust after ~7sec, refresh every 20-30sec.

### [Boulderfist Hunter](https://tbc.wowhead.com/npc=18352/boulderfist-hunter)
- Uses Spear Throw on aggro, recasts every 20-25sec.

### [Gava'xi](https://tbc.wowhead.com/npc=18298/gavaxi)
- Uses Gouge every 30-40sec.
- Uses Sinister Strike in combat. Seems semi-random. maybe every 2-10sec. 
- Uses Eviscerate every 15-30sec.

### [Aged Clefthoof](https://tbc.wowhead.com/npc=17133/aged-clefthoof)
- Uses Rushing Charge only if player is out of melee range. Did not use on aggro, but when being kited it used it after ~10sec and reused every ~30sec.

### [Lake Surger](https://tbc.wowhead.com/npc=17155/lake-surger)
- Uses Surge on Aggro, recasts every 30-40sec.

### [Murkblood Putrifier](https://tbc.wowhead.com/npc=18202/murkblood-putrifier)
- Uses Corrupted Earth every 10-15sec.
- Uses Tainted Chain Lightning every 15sec.
- At <40% HP, spams Putrid Cloud and nothing else.

### [Murkblood Scavenger](https://tbc.wowhead.com/npc=18207/murkblood-scavenger)
- No Observable Mechanics.

### [Murkblood Raider](https://tbc.wowhead.com/npc=18203/murkblood-raider)
- Uses Sunder every 15sec.
- Uses Cleave frequently in combat. Maybe every 10-20sec.

### [Gan'arg Tinkerer](https://tbc.wowhead.com/npc=17151/ganarg-tinkerer)
- Honestly I don't know. They DO use Power Burn, but not consistent. Some used it once in their life span, others never did. Killed one slowly over the course of about 3min and it was never used, but when I pulled a few of them and bursted, combat log showed Power Burn used by a couple of them.  

### [Felguard Legionnaire](https://tbc.wowhead.com/npc=17152/felguard-legionnaire)
- Uses Fel Fire, Cleave, and Cutdown all in combat, all quite frequently. No HP% required for any of them to trigger. 
- This was hard to test because this f*cking thing destroys me in seconds.  

### [Mountain Gronn](https://tbc.wowhead.com/npc=19201/mountain-gronn)
- No Observable Mechanics

### [Greater Windroc](https://tbc.wowhead.com/npc=17129/greater-windroc)
- No Observable Mechanics

### [Mo'arg Engineer](https://tbc.wowhead.com/npc=16945/moarg-engineer)
- Never used Magnetic Pull. Even while I kited it with frost shock for 5 minutes. Wowhead says he has this ability however so I don't know. Maybe I was unlucky. 
- Thorium Drill and Puncture used periodically during combat if in melee range. 

### [Demos, Overseer of Hate](https://tbc.wowhead.com/npc=18535/demos-overseer-of-hate)
- Does nothing but slap me with melee as hard as a damn truck. 
- This was difficult to test because he just destroys me. Only tested for about 1.4min and only got him to 90% hp.

### [Xirkos, Overseer of Fear](https://tbc.wowhead.com/npc=18536/xirkos-overseer-of-fear)
- No Observable Mechanics. 
- Fought him for about 2min in melee range, then proceeded to kite him for 18minutes through Nagrand, into Zangarmarsh, and all across Zangarmarsh because he runs slow and I could. Just taking my legion commander for a walk.  

### [Talbuk Patriarch](https://tbc.wowhead.com/npc=18226/talbuk-patriarch)
- Doesn't seem to exist in game? TBC WoWhead says location unknown. Maybe added later? 

### [Saurfang the Younger ](https://tbc.wowhead.com/npc=18229/saurfang-the-younger)
- I cannot test him as he is friendly to Horde and I have no Alliance characters on Live.

### [Watoosuns Polluted Essence](https://tbc.wowhead.com/npc=18145/watoosuns-polluted-essence)
- No Observable Mechanics

### [Bull Elekk](https://tbc.wowhead.com/npc=18289/bull-elekk)
- Uses Trample every 10sec.

### [Terrorguard](https://tbc.wowhead.com/npc=18661/terrorguard)
- Uses Cripple after ~10-15sec, recasts every ~40sec.
- Uses Warstomp every ~10-20sec

### [Subjugator Vaz'shir](https://tbc.wowhead.com/npc=18660/subjugator-vazshir)
- No Observable Mechanics.

### [Agitated Orc Spirit](https://tbc.wowhead.com/npc=18043/agitated-orc-spirit)
- No Observable Mechanics

### [Twilight Serpent](https://tbc.wowhead.com/npc=23026/twilight-serpent)
- No Observable Mechanics

### [Warmaul Brute](https://tbc.wowhead.com/npc=18065/warmaul-brute)
- Uppercut after ~10sec. Recast every 15-20sec.

### [Muck Spawn](https://tbc.wowhead.com/npc=17154/muck-spawn)
- Poison bolt in combat every 5-10sec.
- Mucky Sludge was seen rarely. Cast every 20-45sec.

### [Boulderfist Warrior](https://tbc.wowhead.com/npc=17136/boulderfist-warrior)
- Shoulder Charge on aggro. Recast every 50-60sec if target out of melee range. 
- Dualweilds

### [Zorbo the Advisor](https://tbc.wowhead.com/npc=18413/zorbo-the-advisor)
- Uses Frost Shock in combat on CD. Every ~7-10sec.
- Uses Scorching Totem in combat every ~10sec.
- Uses Healing Wave once when HP low. Not sure on exact percentage.

### [Gutripper](https://tbc.wowhead.com/npc=18257/gutripper)
- Uses Gutrip after ~20sec. Refreshes every ~2min 10sec when debuff is expired.

### [Ortor of Murkblood](https://tbc.wowhead.com/npc=18204/ortor-of-murkblood)
- Couldn't test. Too many people killing him with a long respawn time. 

### [Avatar of Sathal](https://tbc.wowhead.com/npc=21925/avatar-of-sathal)
- Couldn't test. 

---

As usual, this is Live TBC information from first hand testing on my level 70 shaman. If anyone has evidence that these mobs changed between TBC and WotLK, please provide it!

**Comment by FrancescoBorzi** [comment]
thanks for the great feedback @wetbrownsauce !

I'd say we can merge this PR and address any leftovers later, do you guys agree?

**Comment by Shybuia** [comment]
@FrancescoBorzi I will just re-test really quickly the Mo'arg master planner and it will be ready to go

**Comment by Shybuia** [comment]
@FrancescoBorzi I retested it and guys were right @wetbrownsauce @Knindzagxg , I Just made mistake in testing, it works properly https://youtu.be/jCmdxTgx994 now just decide to merge as you said, if everyone else agree to

note: sorry, video is laggy, I was compiling app in meanwhile, but it can be seen near the end of video

**Comment by wetbrownsauce** [comment]
I don't know if it was clear, I didn't test this PR. The data I provided is how the mobs function on Live TBC-classic servers, which happens to be entirely different from the way they behave with this PR.  

If it feels improved I suppose it can be merged but it definitely isn't blizzlike.

**Comment by FrancescoBorzi** [comment]
@Knindzagxg can you please check?

**Comment by Knindzagxg** [comment]
Already re-updated with issues that are found in previous testing.
If someone else can point out what and where, i will re-check and update if need.

**Comment by Shybuia** [comment]
As far as I know conclusion is: 
All changes passed, by @wetbrownsauce tbc live feedback this is not 100% blizzlike but we dont exactly know if changes were made after tbc so we can merge this as improvement and re-open in future if needed or if we get any info how npc should exactly behave

**Comment by Knindzagxg** [comment]
Rechecked and lots of things are moved to be hp% based once WOTLK / CATA hit us (Tho in Cata there are extra spells, but none of them added in SQL atm - Removed some few weeks ago tho)

**Comment by Knindzagxg** [comment]
Retested required changes -> Everything working now as suggested.

**Pull request merged**